report.php

```diff
@@ -21,40 +21,88 @@
 require_once($CFG->dirroot . '/admin/tool/log/store/xapi/lib.php');
 require_once($CFG->dirroot . '/admin/tool/log/store/xapi/classes/form/reportfilter_form.php');
 
-$pagename = 'logstorexapierrorlog';
-admin_externalpage_setup($pagename);
+// Resend values.
+define('XAPI_REPORT_RESEND_FALSE', false);
+define('XAPI_REPORT_RESEND_TRUE', true);
 
+// Default values on url parameters.
 define('XAPI_REPORT_STARTING_PAGE', 0);
 define('XAPI_REPORT_PERPAGE_DEFAULT', 40);
-define('XAPI_REPORT_RESEND_FALSE', false);
-define('XAPI_REPORT_RESEND_TRUE', true);
+define('XAPI_REPORT_ONPAGE_DEFAULT', '');
+define('XAPI_REPORT_EVENTCONTEXT_DEFAULT', '0');
+define('XAPI_REPORT_EVENTNAMES_DEFAULT', array());
+define('XAPI_REPORT_ERROTYPE_DEFAULT', '0');
+define('XAPI_REPORT_RESPONSE_DEFAULT', '0');
+define('XAPI_REPORT_USERNAME_DEFAULT', '');
+define('XAPI_REPORT_DATEFROM_DEFAULT', 0);
+define('XAPI_REPORT_DATETO_DEFAULT', 0);
 
 // Set context.
 $systemcontext = context_system::instance();
 
+// Add require login and only admin allowed to see this page.
+require_login(null, false);
+if (!has_capability('moodle/site:config', $systemcontext)) {
+    print_error('accessdenied', 'admin');
+}
+
+// Read parameters
 $id           = optional_param('id', XAPI_REPORT_ID_ERROR, PARAM_INT); // This is the report ID
 $page         = optional_param('page',XAPI_REPORT_STARTING_PAGE, PARAM_INT);
 $perpage      = optional_param('perpage', XAPI_REPORT_PERPAGE_DEFAULT, PARAM_INT);
+$onpage       = optional_param('onpage', XAPI_REPORT_ONPAGE_DEFAULT, PARAM_TEXT);
 
-$baseurl = new moodle_url('/admin/tool/log/store/xapi/report.php', array('id' => $id, 'page' => $page, 'perpage' => $perpage));
+// Set pagination url's parameter.
+$urlparams = array(
+    'id' => $id,
+    'page' => $page,
+    'perpage' => $perpage,
+    'onpage' => $onpage
```

Comment:
Do we really need 2 params for which page the user is on?

Reply:
Yes, we need it.